src/wallet/walletdb.h

```diff
@@ -43,8 +43,16 @@ class CKeyMetadata
 {
 public:
     static const int CURRENT_VERSION=1;
+    static const int VERSION_SUPPORT_FLAGS=2;
+    static const uint8_t KEY_ORIGIN_UNSET         = 0x0000;
+    static const uint8_t KEY_ORIGIN_UNKNOWN       = 0x0001;
+    static const uint8_t KEY_ORIGIN_IMPORTED      = 0x0002;
+    static const uint8_t KEY_ORIGIN_UNENC_WALLET  = 0x0004;
+    static const uint8_t KEY_ORIGIN_ENC_WALLET    = 0x0008;
```

Comment:
The main idea was to allow multiple key flags. Like (`imported`) AND (`encrypted` OR `unencrypted`) or potential new key flags like (`UI` OR `RPC` together with AND's above).
I just thought when changing the database model, do it right and flexible.